lib/Twig/Extension/Core.php

```diff
@@ -1277,7 +1277,15 @@ function twig_lower_filter(Twig_Environment $env, $string)
     function twig_title_string_filter(Twig_Environment $env, $string)
     {
         if (null !== ($charset = $env->getCharset())) {
-            return mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = mb_convert_case($string, MB_CASE_TITLE, $charset);
+            $title = preg_replace('#\'S#u', '\'s', $title);
+
+            return preg_replace_callback(
+                            '#\b\d+([^\d\s]+)\b#u',
+                            function ($matches) {
```

Comment:
I'm not too happy to fix a PHP bug, especially if it means adding too much code in Twig itself. It would be much better to fix PHP instead.